source/snapshot/iterator.go

```diff
@@ -83,14 +85,19 @@ func (i *Iterator) Next(ctx context.Context) (sdk.Record, error) {
 			if err := i.t.Err(); err != nil {
 				return sdk.Record{}, fmt.Errorf("fetchers exited unexpectedly: %w", err)
 			}
+			if err := i.acks.Wait(ctx); err != nil {
```

Comment:
Curious what are we guarding against here? Potentially Conduit calling next without saving the previous record?

Reply:
We're guarding against shutting down the pipeline before all records are processed. Since we don't have long polling right now, the pipeline will stop with an error saying `snapshot complete`, but we don't want that error to be emitted until we received all acks, otherwise some records might get dropped.